site/docs/serialization/binary_serialization.md

```diff
@@ -1,5 +1,63 @@
 # Binary Serialization
 
+Substrait can be serialized into a [protobuf](https://developers.google.com/protocol-buffers) based binary representation. The proto schema/idl files can be found on [GitHub](https://github.com/substrait-io/substrait/tree/main/binary). Proto files are place in the `io.substrait` namespace for C++/Java and the `Substrait.Protobuf` namespace for C#. 
+
+
+## Plan
+
+The main top-level object used to communicate a Substrait plan using protobuf is a Plan message. The plan message is composed of a set of data structures that minimize repetition in the serialization along with one (or more) Relation trees. 
+
+=== "Plan Message"
+
+    ```proto
+%%% proto.message.Plan %%%
+    ```
+
+## Extensions
+Protobuf supports both [simple](/extensions/#simple-extensions) and [advanced](/extensions/#advanced-extensions) extensions. Simple extensions are declared at the plan level and advanced extensions are declared at multiple levels of messages within the plan.
+
+### Simple Extensions
+
+For simple extensions, a plan references the URIs associated with the simple extensions to provide additional plan capabilities. These URIs will list additional relevant information for the plan. 
+
+Since references to simple extensions are used extensively in most plans, the serialized representation is normalized within the plan message type minimize space and serialization costs. For each simple extension YAML referenced, the plan will defined an ExtensionURI object and in-plan surrogate key that can be used in each simple extension extension definition. 
```

Comment:
```suggestion
For simple extensions, a plan references the URIs associated with the simple extensions to provide additional plan capabilities. These URIs will list additional relevant information for the plan.  For each simple extension YAML referenced, the plan will define an ExtensionURI object and in-plan surrogate key which is used to define specific extensions. 
```

Reply:
Did you not like the second section? Everyone who has encountered this pattern has seem so confused by the pattern than I added this section. Or do you think it was just covered well enough elsewhere?

Reply:
I think the section is ok.  I proposed removing the sentence "Since references to simple extensions are used extensively in most plans, the serialized representation is normalized within the plan message type minimize space and serialization costs."

> Since references to simple extensions are used extensively in most plans

Conjecture?  Also, might worry/confuse people that aren't using simple extensions when they get started.

> the serialized representation is normalized within the plan message type minimize space and serialization costs.

Typo, and also the rationale could be confusing.  I think once people understand the "what" the "why" will be pretty obvious.  I'd be fine with this part coming back as I could be convinced the "why" helps in understanding the "what".

Reply:
> Conjecture? 

Since all functions are considered simple extensions, it seems like everyone will be using them (unless you make a plan with zero functions). I'm not sure what is the clear way to say that. Thoughts?

Reply:
Ah, I see.  Perhaps we need to pick some terms.

"Surrogate key" is jargony although maybe apropos for databases and "pointer" feels a bit out of place.  Elsewhere in the spec/serialization domain this concept would be called referneces ([JSON](https://json-spec.readthedocs.io/reference.html)), key/keyref ([XML](https://docs.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/ms256478(v=vs.100))), anchor/reference ([YAML](https://support.atlassian.com/bitbucket-cloud/docs/yaml-anchors/)).  So I think there is some agreement on "reference" at least and that is the term I would expect.

That being said, you already use "reference" for the name of the first section but maybe "declaration" would be a better term.  So something like...

```
Extensions within a plan are split into two parts, a declaration and a number of references.  The declaration is encapsulated in the top level `extensions` section of the plan.  An extension declaration includes a URI which should be a unique identifier for the extension and should ideally point to a YAML document describing the extension.  An extension declaration also includes a key, this is a plan-specific value that the producer creates as an anchor/key to be referenced elsewhere.

An extension declaration is then referenced throughout the body of the plan itself.  The exact structure of these references will depend on the extension point being used but they will always include the extension's key.  For example, all scalar functions contain references to an extension which defines the semantics of the function.
```

Reply:
I did a pass based on your ideas/text. LMK your thoughts.